crates/invoker-impl/src/lib.rs

```diff
@@ -504,7 +506,17 @@ where
                 endpoint_id,
                 ism.invocation_state_debug()
             );
-            ism.notify_chosen_endpoint(endpoint_id);
+
+            self.status_store.on_endpoint_chosen(
+                &partition,
+                &full_invocation_id,
+                endpoint_id.clone(),
+            );
+            // If we think this selected endpoint has been freshly picked, otherwise
+            // we assume that we have stored it previously.
+            if has_changed {
+                ism.notify_chosen_endpoint(endpoint_id);
+            }
```

Comment:
Can we have a unit test for this `has_changed` behavior?

Reply:
So next time i stumble on it i can figure it out from the tests.

Reply:
It's a good idea but I'm not sure if this is going to be an easy to slot in existing tests. I'm a bit swamped and it'll be hard now to add a new test only for this case tbh.